oktapam/resource_ad_task_settings.go

```diff
@@ -30,16 +34,11 @@ func resourceADTaskSettings() *schema.Resource {
 				Required:    true,
 				Description: descriptions.Name,
 			},
-			attributes.ID: {
```

Comment:
I think I might just not understand IDs?

Reply:
ID is very confusing in the terraform. Id is treated as "meta-field" and it has it's own getter/setter. 

(*helper/schema.ResourceData).SetId() method does two things - 

- Set the underlying resource identifier
- Set the resource id attribute. 

Existence of resource identifier determine if resource exists or not and it's mandatory to set it while creating resource. I was not sure what is the purpose of keeping ID as the attribute. Then, I looked at the terraform sdk code and see below comment - 

	// once we transition away from the legacy state types, "id" will no longer
	// be a special field, and will become a normal attribute.

Reference: https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/resource_data.go#L286

I can make a change and keep the Id attribute.

Reply:
Ah so if I understand correctly, us humans should not really be using "id" - at least not for project, since if we want the project's id we should use `project_id`, so having an `id` field too is confusing because it's actually the project's name?

Reply:
Sorry, my comment wasn't very clear. I was just trying to explain how "Id" attribute work in terraform and how it's going to change over time and not treated as "meta-field".This PR was created before putting thoughts around how we want to populate tf resource id, so at that time I created extra attribute "project_id" to make tests work. Now, I feel there is no need to introduce new attribute and we can just populate existing id attribute. 

My proposal is to populate "Id" as ASA resource UUID wherever it exist (for plural datasource there is no ASA resource Id). ASA resource UUID is unique and stable attribute. We don't need to base our decision to populate "id" on how read resource api works. We can change tf resource read method and import method to use other required attributes instead of id to retrieve resource back. @stevenelleman-okta want to review this approach first with you before making changes all over. 

I think better to talk on this once you come online.

Reply:
Fixed in [6c0355a](https://github.com/okta/terraform-provider-oktapam/pull/65/commits/6c0355a9175e980f2658c6d23769aa347eac8d8f)